sdkv2/helper.go

```diff
@@ -159,9 +159,9 @@ var (
 )
 
 const (
-	// KeyRouteAffinityWrite enables routing optimization for all write operations
+	// KeyRouteAffinityWrite enables routing optimization for conditional write operations
```

Comment:
Nitpick about the word "optimization" in this comment and in several other places below (like the boolean "shouldOptimize"). The "key route affinity" - basically choosing where to send the request based just on the **partition key** (not the whole key, by the way!!!) - is hard to call "optimization". It doesn't speed up the query. It might even make it slower and more expensive (e.g., cross AZ boundary). The only purpose of this feature is to avoid LWT contention.

Reply:
Also, I think the name of these two constants - "Write" to refer to read-modify-write operations and "all" to refer to all **write** operations is misleading.
I would name them "RMW" (read-modify-write, a.k.a. read-before-write) vs purely "write". Alternatively you can call them "lwt writes" vs "all writes".

I guess these names made more sense when the distinction was between writes and reads+writes (all operations). But it's not the case any more.

Reply:
Thanks, I changed naming and comments